legend-engine-xts-analytics/legend-engine-xts-analytics-quality/legend-engine-xt-analytics-quality-api/src/main/java/org/finos/legend/engine/api/analytics/DataspaceQualityAnalytics.java

```diff
@@ -0,0 +1,242 @@
+//  limitations under the License.
+
+package org.finos.legend.engine.api.analytics;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.swagger.annotations.Api;
+import io.swagger.annotations.ApiOperation;
+import org.eclipse.collections.api.RichIterable;
+import org.finos.legend.engine.language.pure.compiler.toPureGraph.PureModel;
+import org.finos.legend.engine.language.pure.modelManager.ModelManager;
+import org.finos.legend.engine.protocol.pure.v1.model.context.PureModelContextData;
+import org.finos.legend.engine.protocol.pure.v1.model.packageableElement.PackageableElement;
+import org.finos.legend.engine.shared.core.ObjectMapperFactory;
+import org.finos.legend.pure.generated.Root_meta_analytics_quality_model_CheckResult;
+import org.finos.legend.pure.generated.Root_meta_pure_functions_collection_List_Impl;
+import org.finos.legend.pure.generated.core_analytics_quality_associationChecks;
+import org.finos.legend.pure.generated.core_analytics_quality_classChecks;
+import org.finos.legend.pure.generated.core_analytics_quality_enumerationChecks;
+import org.finos.legend.pure.generated.core_analytics_quality_functionChecks;
+import org.finos.legend.pure.generated.core_analytics_quality_propertyChecks;
+import org.finos.legend.pure.m3.coreinstance.meta.pure.metamodel.function.ConcreteFunctionDefinition;
+import org.finos.legend.pure.m3.coreinstance.meta.pure.metamodel.function.property.Property;
+import org.finos.legend.pure.m3.coreinstance.meta.pure.metamodel.type.Class;
+import org.finos.legend.pure.m3.coreinstance.meta.pure.metamodel.type.Enum;
+import org.finos.legend.pure.m3.coreinstance.meta.pure.metamodel.type.Enumeration;
+import org.finos.legend.pure.runtime.java.compiled.execution.CompiledExecutionSupport;
+
+import javax.ws.rs.Consumes;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Api(tags = "Analytics - Quality")
+@Path("pure/v1/analytics/quality")
+public class DataspaceQualityAnalytics
+{
+    private final ModelManager modelManager;
+    private List<String> messages;
+
+    public DataspaceQualityAnalytics(ModelManager modelManager)
+    {
+        this.modelManager = modelManager;
+    }
+
+    @POST
+    @Path("check")
+    @ApiOperation("Checks the quality of provided Data space")
+    @Consumes(MediaType.APPLICATION_JSON)
+    public Response checkDataSpaceConstraints(PureModelContextData pureModelContextData)
+    {
+        messages = new ArrayList<>();
+        PureModel pureModel = this.modelManager.loadModel(pureModelContextData, null, null, null);
+
+        List<PackageableElement> classes = pureModelContextData.getElements().stream().filter(e -> e instanceof org.finos.legend.engine.protocol.pure.v1.model.packageableElement.domain.Class).collect(Collectors.toList());
```

Comment:
should we be just compiling the dataspcae and then call a pure function for "check constraints" on that dataspace, rather than embedding the logic in what is an API class, which ideally has no logic at all